pkg/gardenlet/controller/shoot/shoot_control_delete.go

```diff
@@ -545,6 +545,11 @@ func (r *shootReconciler) runDeleteShootFlow(ctx context.Context, o *operation.O
 			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.KubeAPIServerService.Destroy).RetryUntilTimeout(defaultInterval, defaultTimeout),
 			Dependencies: flow.NewTaskIDs(deleteKubeAPIServer, destroyKubeAPIServerSNI),
 		})
+		_ = g.Add(flow.Task{
+			Name:         "Destroying gardener-resource-manager",
+			Fn:           flow.TaskFn(botanist.Shoot.Components.ControlPlane.ResourceManager.Destroy),
+			Dependencies: flow.NewTaskIDs(deleteKubeAPIServer),
+		})
```

Comment:
Wait a sec, doesn't the `Deleting referenced resources` step require gardener-resource-manager to be able to delete the `referenced-resources` ManagedResource? If you destroy GRM here, will the deletion of the `referenced-resources` MR succeed?

Reply:
Too many non-obvious cases :) Probably we should extend the integration test with using referenced resources to cover this case?

Reply:
Ah, I see that `referenced-resources` MR is with class `seed`. So I think the central GRM should take care of its deletion.

Reply:
/unhold

Reply:
Just looking at the flow, the dependency graph is:
```
DestroyGardenerResourceManager <- deleteKubeAPIServer <- syncPointCleaned <- waitUntilManagedResourcesDeleted
```

I also had the chance today to test the change and indeed the only managed resource left is `referenced-resources` but it is managed by the seed resource manager.